src/mqt/benchviewer/backend.py

```diff
@@ -553,10 +441,185 @@ def handle_downloading_benchmarks(self, target_location: str, download_url: str)
                 bar.update(size)
         print(f"Download completed to {fname}. Server is starting now.")
 
-        return
 
+def parse_data(filename: str) -> ParsedBenchmarkName:
+    """Extracts the necessary information from a given filename.
+
+    Keyword arguments:
+    filename -- name of file
+
+    Return values:
+    parsed_data -- parsed data extracted from filename
+    """
+    benchmark = filename.split("_")[0].lower()
+    num_qubits = get_num_qubits(filename)
+    indep_flag = "indep" in filename
+    nativegates_flag = "nativegates" in filename
+    mapped_flag = "mapped" in filename
+    compiler, compiler_settings = get_compiler_and_settings(filename)
+    gate_set = get_gate_set(filename) if nativegates_flag or mapped_flag else None
+    target_device = get_target_device(filename) if mapped_flag else None
+
+    return ParsedBenchmarkName(
+        benchmark=benchmark,
+        num_qubits=num_qubits,
+        indep_flag=indep_flag,
+        nativegates_flag=nativegates_flag,
+        mapped_flag=mapped_flag,
+        compiler=compiler,
+        compiler_settings=compiler_settings,
+        gate_set=gate_set,
+        target_device=target_device,
+        filename=filename,
+    )
+
+
+class NoSeekBytesIO:
+    def __init__(self, fp: io.BytesIO) -> None:
+        self.fp = fp
+        self.deleted_offset = 0
+
+    def write(self, b: bytes) -> int:
+        return self.fp.write(b)
+
+    def tell(self) -> int:
+        return self.deleted_offset + self.fp.tell()
 
-def get_tket_settings(filename: str):
+    def hidden_tell(self) -> int:
+        return self.fp.tell()
+
+    def seekable(self) -> bool:
+        return False
+
+    def hidden_seek(self, offset: int, start_point: int = io.SEEK_SET) -> int:
+        return self.fp.seek(offset, start_point)
+
+    def truncate_and_remember_offset(self, size: int | None) -> int:
+        self.deleted_offset += self.fp.tell()
+        self.fp.seek(0)
+        return self.fp.truncate(size)
+
+    def get_value(self) -> bytes:
+        return self.fp.getvalue()
+
+    def close(self) -> None:
+        return self.fp.close()
+
+    def read(self) -> bytes:
+        return self.fp.read()
+
+    def flush(self) -> None:
+        return self.fp.flush()
+
+
+def parse_benchmark_id_from_form_key(k: str) -> int | None:
+    pat = re.compile(r"_\d+")
+    m = pat.search(k)
+    if m:
+        return int(m.group()[1:])
+    return None
+
+
+def prepare_form_input(form_data: dict[str, str]) -> BenchmarkConfiguration:
```

Comment:
Adding a test for this function wouldn't hurt probably.

Reply:
Extended the test for that.